addons/l10n_in/models/company.py

```diff
@@ -12,3 +11,54 @@ class ResCompany(models.Model):
         ],
         string="HSN Code Digit",
     )
+
+    def write(self, vals):
+        old_state_vals = {c.id: c.state_id.id for c in self}
+        res = super().write(vals)
+        if vals.get('state_id'):
+            # Update Fiscal Positions for companies setting up state for the first time
+            companies_need_update_fp = self.filtered(lambda c:\
+                c.account_fiscal_country_id.code == 'IN' and\
+                not old_state_vals.get(c.id) and\
+                c.state_id and\
+                c.state_id.country_id.code == 'IN')
+            for company in companies_need_update_fp:
+                company._update_l10n_in_fiscal_position()
+        return res
+
+    def _get_l10n_in_fiscal_positions_vals_by_ref(self):
+        self.ensure_one()
+        return {
+            'fiscal_position_in_intra_state': {
+                'sequence': 1,
+                'auto_apply': True,
+                'country_id': self.country_id.id,
+                'state_ids': [Command.set(self.state_id.ids)]
+            },
+            'fiscal_position_in_inter_state': {
+                'sequence': 2,
+                'auto_apply': True,
+            },
```

Comment:
Why is this needed?
```suggestion
```

Reply:
May be we can keep a empty dict? it might be useful for setting up branches (if we may move sequence and auto_apply to csv as you suggested) and easy to update vals in future if required